src/plugins/discover/public/application/main/components/sidebar/discover_sidebar_responsive.tsx

```diff
@@ -252,23 +252,26 @@ export function DiscoverSidebarResponsive(props: DiscoverSidebarResponsiveProps)
   }, []);
 
   const { dataViewFieldEditor, dataViewEditor } = services;
-  const { availableFields$ } = props;
+  // const { availableFields$ } = props;
 
   const canEditDataView =
     Boolean(dataViewEditor?.userPermissions.editDataView()) || !selectedDataView?.isPersisted();
 
+  /**
   useEffect(() => {
     // For an external embeddable like the Field stats
     // it is useful to know what fields are populated in the docs fetched
     // or what fields are selected by the user
 
+    // todo this needs to be removed
     const availableFields =
       props.columns.length > 0 ? props.columns : Object.keys(sidebarState.fieldCounts || {});
     availableFields$.next({
       fetchStatus: FetchStatus.COMPLETE,
       fields: availableFields,
     });
   }, [selectedDataView, sidebarState.fieldCounts, props.columns, availableFields$]);
+*/
```

Comment:
We should do this step by step. So first, let's get rid of `availableFields$`. It's currently just used to collect the fields that are available in Discover's sidebar. And its only consumer is the field statistics table:

https://github.com/elastic/kibana/blob/22d0fa742df4b01e31a838419627b0583b5a4d83/src/plugins/discover/public/application/main/components/field_stats_table/field_stats_table.tsx#L157-L167

So I think a first feasible goal could be to migrate the fetching of fields to the only component that is actually is using it. This would further decouple the code.